src/lib.rs

```diff
@@ -234,6 +238,14 @@ fn get_num_cpus() -> usize {
 fn get_num_cpus() -> usize {
     use std::ptr;
 
+    #[cfg(target_os = "freebsd")]
+    {
+        let cpus = get_cpuset_cpus();
+        if cpus > 0 {
+            return cpus;
+        }
```

Comment:
Is it on purpose that this falls through to the original function?

Reply:
I mirrored the Linux logic, which includes a sysconf() fallback if sched_getaffinity() fails for some reason. I have no particular concern that cpuset_getaffinity() will fail any more so than sched_getaffinity() looks likely to fail, I just erred on the side of caution because it was easy.

Reply:
I suppose the main concern with both paths is if cpuset_t changes and it starts returning ERANGE because the set we're providing is now too small.